### PR TITLE
Issue #24 Add the ability to generate the quarkus reflection-config.jon file

### PR DESCRIPTION
**Pull request by averabaq**
Issue #24 Add the ability to generate the quarkus reflection-config.jon file

### CODE REVIEW

File: src/main/kotlin/com/cjbooms/fabrikt/generators/model/QuarkusReflectionModelGenerator.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.SerializationFeature
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+
+class QuarkusReflectionModelGenerator(
+    private val models: Models,
+    private val options: Set<ModelCodeGenOptionType> = emptySet()
+) {
+    companion object {
+        const val RESOURCE_FILE_NAME = "reflection-config.json"
```
Comment (cjbooms) [inline src/main/kotlin/com/cjbooms/fabrikt/generators/model/QuarkusReflectionModelGenerator.kt:16]: The problem with having a hard-coded file name in `/src/main/resources` is that multiple generate tasks will overwrite each other

File: README.md [hunk 2]
```diff
@@ -39,6 +39,7 @@ This section documents the available CLI parameters for controlling what gets ge
  |   `--http-model-opts`  | Select the options for the http models that you want to be generated.
  |                        | CHOOSE ANY OF:
  |                        |   `JAVA_SERIALIZATION` - This option adds Java Serializable interface to the generated models
+ |                        |   `QUARKUS_REFLECTION_CONFIG` - This options generates the reflection-config.json file for quarkus integration projects
```
Comment (cjbooms) [inline README.md:42]: What about splitting this into two options:
`--http-model-opts=QUARKUS_REFLECTION` This option adds `@RegisterForReflection` to the generated models. Requires dependency "'io.quarkus:quarkus-core:+"

And a separate target for generating the json reflection:
`--targets= QUARKUS_REFLECTION_CONFIG` This options generates the reflection-config.json file for quarkus integration projects

Comment (averabaq) [reply README.md:42]: 👍

Comment (averabaq) [reply README.md:42]: sold!

Comment (cjbooms) [reply README.md]: 🥳

**Comment by cjbooms** [comment]
👍

**Pull request merged**